tasks/Task.cpp

```
@@ -3,31 +3,39 @@
 #include "Task.hpp"
 
 using namespace pipeline_follower;
+using namespace auv_trajectory_control;
+
 
 Task::Task(std::string const& name)
     : TaskBase(name)
 {
+    currentVelocity = 0.0;
 }
 
 Task::Task(std::string const& name, RTT::ExecutionEngine* engine)
     : TaskBase(name, engine)
 {
+    currentVelocity = 0.0;
 }
 
 Task::~Task()
 {
 }
 
-
-
-/// The following lines are template definitions for the various state machine
-// hooks defined by Orocos::RTT. See Task.hpp for more detailed
-// documentation about them.
-
+bool Task::setSegment_length(double value)
+{
+    segment_length = std::abs(value);
+    return(pipeline_follower::TaskBase::setSegment_length(value));
+}
 bool Task::configureHook()
 {
     if (! TaskBase::configureHook())
         return false;
+
+    segment_length = std::abs(_segment_length.get());
```

Don't silently accept invalid configuration. Check if they are positive, and refuse configuration if they're not. Same in setSegment_length